### PR TITLE
Remove deprecated compatibility code.

### PR DESCRIPTION
**Pull request by bnoordhuis**
* Remove support for the IOJS_ORG_MIRROR, NVM_IOJS_ORG_MIRROR
  and NVM_NODEJS_ORG_MIRROR enviroment variables.

* Remove obsolete support for io.js. It's been out of support
  for over three years now.

### CODE REVIEW

File: test/docker.sh [hunk 1]
```diff
@@ -3,7 +3,6 @@
 #set -e
 
 test_node_versions="0.8.28 0.10.40 0.12.7 4.3.0 5.6.0"
```
Comment (richardlau) [inline test/docker.sh]: Should we be updating these (perhaps in a different PR) to in support versions of Node.js?

Comment (bnoordhuis) [reply test/docker.sh]: Yep, good idea. I could do that as a separate commit or a separate PR.

Comment (richardlau) [reply test/docker.sh]: > 
> 
> Yep, good idea. I could do that as a separate commit or a separate PR.

Your choice 😄

**Comment by bnoordhuis** [comment]
Updated after some delay... CI: https://ci.nodejs.org/view/All/job/nodegyp-test-pull-request/116/

**Pull request closed without merging**